manifests/proxydns.pp

```diff
@@ -21,9 +21,29 @@
     soaip   => $ip,
   }
 
-  ::dns::zone { $foreman_proxy::dns_reverse:
-    soa     => $::fqdn,
-    reverse => true,
-    soaip   => $ip,
+  $foreman_proxy::dns_reverse.each |String $reverse| {
+    dns::zone { $reverse:
+      soa     => $::fqdn,
+      reverse => true,
+      soaip   => $ip,
+    }
+    @@dns::zone { "${::fqdn}_reverse_${reverse}":
+      zonetype    => 'slave',
+      masters     => [ $ip ],
+      manage_file => false,
+      soaip       => $ip,
+      zone        => $reverse,
+    }
+  }
+
+  @@dns::zone { "${::fqdn}_forward_slave":
+    zonetype    => 'slave',
+    masters     => [ $ip ],
+    manage_file => false,
+    soaip       => $ip,
+    zone        => $foreman_proxy::dns_zone,
   }
+
+  Dns::Zone <<| soaip != $ip |>>
```

Comment:
This seems a little broad - it'll collect _every_ zone, whether or not it's meant to be deployed on this server. Should it be a little more targeted at which server is the slave?